app/views/events/show.html.erb

```diff
@@ -4,8 +4,32 @@
 <span>Event name:</span>
 <span><%= @event.name %></span><br>
 <span>Date:</span>
-<span><%= @event.date %></span><br>
+<span>
+  <%= @event.date %> <%= time_ago_in_words(@event.date) %>
+  <%= @event.date > Time.now ? "from now" : "ago" %>
+</span>
+<br>
 <span>Content:</span>
 <span><%= @event.content %></span><br>
 <span>Location:</span>
-<span><%= @event.location %></span><br><br> 
\ No newline at end of file
+<span><%= @event.location %></span><br><br>
+
+<% unless (session[:user_id] == @event.creator_id) || (@event.date < Time.now) %>
```

Comment:
- [ ] There should be absolutely NO logic in views. --  Rails has built-in mechanism to deal with view-related logic which is View Helpers (app/helpers/*_helper.rb). So the common convention is to move any logic in the views to the helpers. Please take a look at [this](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers) for more information.

![](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png)